NightscoutServiceKit/NightscoutService.swift

```diff
@@ -331,20 +331,30 @@ extension NightscoutService: RemoteDataService {
         uploader.uploadProfiles(stored.compactMap { $0.profileSet }, completion: completion)
     }
     
-    public func validatePushNotificationSource(_ notification: [String: AnyObject]) -> Bool {
-        guard let otpToValidate = notification["otp"] as? String else {
-            log.error("Missing OTP")
-            return false
+    public func validatePushNotificationSource(_ notification: [String: AnyObject]) -> Result<Void, Error> {
+        
+        guard let password = notification["otp"] as? String else {
+            return .failure(NotificationValidationError.missingOTP)
+        }
+        
```

Comment:
The date the notification was delivered from Nightscout will be presented in a user message for expired OTPs.